### DISCUSSION TITLE
It would be great if you can include the Azure Linux VM Agent (https://learn.microsoft.com/en-us/azure/virtual-machines/extensions/agent-linux) in your azure images

### DISCUSSION THREAD

**Discussion by NeenaKunnath** [Genymobile/feedback#42] [Genymotion Device Image] [answered]
It would be great if you can include the Azure Linux VM Agent (https://learn.microsoft.com/en-us/azure/virtual-machines/extensions/agent-linux) in your azure images so that users can do all azure operations seemlessly.

**Comment by guillaume-roche** [comment, marked as answer]
Hello,

Thanks for your feedback.

As you know, our Azure images run Android, and it would be rather complex to embed the Linux VM Agent.

We would have to solve at least the following issues:
* The VM Agent is written in Python, and there is no built-in interpreter in Android, so we would need to port or integrate one.
* Some Linux concepts are either not available in Android, or work in a different way (e.g. users). This means that we would have to maintain our own fork of the agent. Some functionalities would probably have to be disabled when they have no purpose in an Android context. Also, such a fork might require specific code depending on the Android version.
* We would have to make it work with the Android SSL stack (BoringSSL) and maybe other dependencies could cause some trouble too.

Therefore, the amount of necessary work seems rather large, and we have other priorities right now. We are keeping this topic in mind, but we don't guarantee that we will be able to handle it.

Best regards;
